cvxpylayers/torch/cvxpylayer.py

```diff
@@ -172,63 +169,60 @@ def forward(ctx, *params):
                         (i, str(ctx.device), str(p.device))
                     )
 
-            # check shapes of params
+            ctx.batch_sizes = []
+            for i, (p, q) in enumerate(zip(params, param_order)):
+                # check and extract the batch size for the parameter
+                # 0 means there is no batch dimension for this parameter
+                # and we assume the batch dimension is non-zero
+                if p.ndimension() == q.ndim:
+                    batch_size = 0
+                elif p.ndimension() == q.ndim + 1:
+                    batch_size = p.size(0)
+                    assert batch_size > 0
+                else:
+                    raise RuntimeError(
```

Comment:
[ValueError](https://docs.python.org/3/library/exceptions.html#ValueError), please :) ValueErrors are informative (it means the user passed in something with the right type but an unacceptable value)

Reply:
updated